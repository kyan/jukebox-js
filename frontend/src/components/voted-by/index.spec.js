import React from 'react'
import { shallow } from 'enzyme'
import VotedBy from './index'

const vote1 = {
  user: {
    _id: '123',
    fullname: 'Big Rainbowhead',
    picture: 'link/to/image'
  },
  at: '2019-12-17T13:11:37.316Z',
  vote: 22
}

const vote2 = {
  user: {
    _id: '999',
    fullname: 'Duncan',
    picture: 'link/to/duncan/image'
  },
  at: '2019-12-19T13:11:37.316Z',
  vote: 75
}

describe('VotedBy', () => {
  let wrapper

  describe('when no votes are provided', () => {
    it('does nothing', () => {
      wrapper = shallow(<VotedBy total={21} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when votes information is provided', () => {
    const votes = [vote1, vote2]

    describe('and no ribbon prop', () => {
      it('displays the correct information', () => {
        wrapper = shallow(<VotedBy total={22} votes={votes} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('and a ribbon prop is set', () => {
      it('displays the correct information', () => {
        wrapper = shallow(<VotedBy total={11} votes={votes} ribbon />)
        expect(wrapper.find('Popup')).toMatchSnapshot()
      })
    })
  })

  describe('when no user data', () => {
    it('does nothing', () => {
      const vote = {
        at: '2019-12-19T13:11:37.316Z',
        vote: 75
      }
      wrapper = shallow(<VotedBy total={10} votes={[vote]} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the vote count is 0', () => {
    it('does not show a label', () => {
      const votes = [
        { at: '2019-12-14T13:11:37.316Z', vote: 0 },
        { at: '2019-12-19T13:11:37.316Z', vote: 0 }
      ]
      wrapper = shallow(<VotedBy total={5} votes={votes} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
