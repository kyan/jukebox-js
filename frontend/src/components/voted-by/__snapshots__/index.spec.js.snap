// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VotedBy when no user data does nothing 1`] = `
<Popup
  content={
    <List>
      <ListItem>
        <ListContent>
          <ListDescription>
            <strong>
              19 Dec 2019 @ 1:11 pm
            </strong>
             - 
            User unknown
             
            <Label
              circular={true}
              size="mini"
            >
              3
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
    </List>
  }
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <Label
      className="track-label vote-track-label"
      color="red"
      content={-4}
      icon="thumbs down"
      ribbon={null}
      size="tiny"
    />
  }
  wide="very"
/>
`;

exports[`VotedBy when no votes are provided does nothing 1`] = `
<Label
  className="track-label vote-track-label"
  color="red"
  content={-3}
  icon="thumbs down"
  ribbon={null}
  size="tiny"
/>
`;

exports[`VotedBy when the vote count is 0 does not show a label 1`] = `
<Popup
  content={
    <List>
      <ListItem>
        <ListContent>
          <ListDescription>
            <strong>
              14 Dec 2019 @ 1:11 pm
            </strong>
             - 
            User unknown
             
            <Label
              circular={true}
              size="mini"
            >
              -5
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
      <ListItem>
        <ListContent>
          <ListDescription>
            <strong>
              19 Dec 2019 @ 1:11 pm
            </strong>
             - 
            User unknown
             
            <Label
              circular={true}
              size="mini"
            >
              -5
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
    </List>
  }
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <Label
      className="track-label vote-track-label"
      color="red"
      content={-4}
      icon="thumbs down"
      ribbon={null}
      size="tiny"
    />
  }
  wide="very"
/>
`;

exports[`VotedBy when votes information is provided and a ribbon prop is set displays the correct information 1`] = `
<Popup
  content={
    <List>
      <ListItem>
        <Image
          as="img"
          avatar={true}
          className="voted_by_avatar_image"
          src="link/to/image"
          ui={true}
        />
        <ListContent>
          <ListDescription>
            <strong>
              17 Dec 2019 @ 1:11 pm
            </strong>
             - 
            Big Rainbowhead
             
            <Label
              circular={true}
              size="mini"
            >
              -3
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
      <ListItem>
        <Image
          as="img"
          avatar={true}
          className="voted_by_avatar_image"
          src="link/to/duncan/image"
          ui={true}
        />
        <ListContent>
          <ListDescription>
            <strong>
              19 Dec 2019 @ 1:11 pm
            </strong>
             - 
            Duncan
             
            <Label
              circular={true}
              size="mini"
            >
              3
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
    </List>
  }
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <Label
      className="track-label vote-track-label"
      color="red"
      content={-4}
      icon="thumbs down"
      ribbon="right"
      size="tiny"
    />
  }
  wide="very"
/>
`;

exports[`VotedBy when votes information is provided and no ribbon prop displays the correct information 1`] = `
<Popup
  content={
    <List>
      <ListItem>
        <Image
          as="img"
          avatar={true}
          className="voted_by_avatar_image"
          src="link/to/image"
          ui={true}
        />
        <ListContent>
          <ListDescription>
            <strong>
              17 Dec 2019 @ 1:11 pm
            </strong>
             - 
            Big Rainbowhead
             
            <Label
              circular={true}
              size="mini"
            >
              -3
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
      <ListItem>
        <Image
          as="img"
          avatar={true}
          className="voted_by_avatar_image"
          src="link/to/duncan/image"
          ui={true}
        />
        <ListContent>
          <ListDescription>
            <strong>
              19 Dec 2019 @ 1:11 pm
            </strong>
             - 
            Duncan
             
            <Label
              circular={true}
              size="mini"
            >
              3
            </Label>
          </ListDescription>
        </ListContent>
      </ListItem>
    </List>
  }
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <Label
      className="track-label vote-track-label"
      color="red"
      content={-3}
      icon="thumbs down"
      ribbon={null}
      size="tiny"
    />
  }
  wide="very"
/>
`;
